includes/Helpers/Transient.php

```diff
@@ -34,12 +34,25 @@ public static function get( string $key ) {
 		}
 
 		/**
-		 * @var array{value:mixed, expires_at:int} $data The saved value and the Unix time it expires at.
+		 * Implement the filters as used in {@see get_transient()}.
+		 */
+		$pre = apply_filters( "pre_transient_{$key}", false, $key );
+		if ( false !== $pre ) {
+			return $pre;
+		}
+
+		/**
+		 * The saved value and the Unix time it expires at.
+		 *
+		 * @var array{value:mixed, expires_at:int} $data
 		 */
 		$data = \get_option( $key );
 		if ( is_array( $data ) && isset( $data['expires_at'], $data['value'] ) ) {
 			if ( $data['expires_at'] > time() ) {
-				return $data['value'];
+				/**
+				 * Implement the filters as used in {@see get_transient()}.
+				 */
+				return apply_filters( "transient_{$key}", $data['value'], $key );
```

Comment:
In the \get_transient function, the filter `transient_{$transient}`is applied even if the transient has expired (see [\get_transient](https://github.com/WordPress/WordPress/blob/fbcb0843059cb1f80edb8826d8ae82e87f1a98b3/wp-includes/option.php#L1482))

I would refactor all return points to store value into a variable instead and return filtered content of the variable at the very end of the method.

The `Transient::get` method would end up looking like this

```php
public static function get( string $key ) {
	if ( self::should_use_transients() ) {
		return \get_transient( $key );
	}

	/**
	 * Implement the filters as used in {@see get_transient()}.
	 */
	$pre = apply_filters( "pre_transient_{$key}", false, $key );
	if ( false !== $pre ) {
		return $pre;
	}

	/**
	 * The saved value and the Unix time it expires at.
	 *
	 * @var array{value:mixed, expires_at:int} $data
	 */
	$data = \get_option( $key );
	if ( is_array( $data ) && isset( $data['expires_at'], $data['value'] ) ) {
		if ( $data['expires_at'] > time() ) {
			$value = $data['value'];
		} else {
			\delete_option( $key );
			$value = false;
		}
	}

	/**
	 * Implement the filters as used in {@see get_transient()}.
	 */
	return apply_filters( "transient_{$key}", $value, $key );
}
```

Reply:
Hi @alarocca130!
I agree with you with reproducing the same behaviour as the one made in the WP Transient, on the last commit I introduced your proposed changes, thanks for the feedback!